core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/generate-certificates.yml

```
@@ -0,0 +1,196 @@
+---
+- name: Generate certificates block
+  vars:
+    # https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates
+    _certificates_opt_mapping:
+      - name: admin.conf
+        kind: ['clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/admin.conf"
+        parent_ca: ca
+      - name: apiserver-etcd-client
+        kind: ['clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/apiserver-etcd-client"
+        parent_ca: etcd/ca
+      - name: apiserver-kubelet-client
+        kind: ['clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/apiserver-kubelet-client"
+        parent_ca: ca
+      - name: apiserver
+        kind: ['serverAuth']
+        target: "{{ specification.advanced.certificates.location }}/apiserver"
+        parent_ca: ca
+      - name: controller-manager.conf
+        kind: ['clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/controller-manager.conf"
+        parent_ca: ca
+      - name: etcd-healthcheck-client
+        kind: ['clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/etcd/healthcheck-client"
+        parent_ca: etcd/ca
+      - name: etcd-peer
+        kind: ['serverAuth', 'clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/etcd/peer"
+        parent_ca: etcd/ca
+      - name: etcd-server
+        kind: ['serverAuth', 'clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/etcd/server"
+        parent_ca: etcd/ca
+      - name: front-proxy-client
+        kind: ['clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/front-proxy-client"
+        parent_ca: front-proxy-ca
+      - name: scheduler.conf
+        kind: ['clientAuth']
+        target: "{{ specification.advanced.certificates.location }}/scheduler.conf"
+        parent_ca: ca
+  block:
+    - name: Create certificates_opt_mapping fact
+      block:
+        - name: Set non-default certificates_opt_mapping fact value
+          set_fact:
+            certificates_opt_mapping: "{{ certificates_opt_mapping | default([]) + [item] }}"
+          when:
+            - certificates_renewal_list is defined
+            - item.name in certificates_renewal_list
+          loop: "{{ _certificates_opt_mapping }}"
+
+        - name: Set default certificates_opt_mapping fact value
+          set_fact:
+            certificates_opt_mapping: "{{ _certificates_opt_mapping }}"
+          when: certificates_renewal_list is not defined
+
+    - name: Save old certificates
+      synchronize:
+        src: "{{ specification.advanced.certificates.location }}/"
+        dest: >-
+          "{{ specification.advanced.certificates.location | regex_replace('\\/$', '') }}-backup-{{ ansible_date_time.iso8601_basic_short }}"
```

```suggestion
          {{ specification.advanced.certificates.location | regex_replace('\\/$', '') }}-backup-{{ ansible_date_time.iso8601_basic_short }}
```
Not sure if you put quotes here by purpose but they are treated literally (preserved) when using inside YAML’s block styles (`>` or `|` ).

---

You're right, it shouldn't be here and was done correctly for 0.5.